addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -113,6 +113,28 @@ def _compute_priority(self):
             if new_priority != batch.priority:
                 batch.priority = new_priority
 
+    @api.multi
+    @api.constrains("picking_ids")
+    def _assign_picks(self):
+        """If configured, attempt to assign all the relevant pickings in self"""
+        if self.env.context.get("lock_batch_state"):
+            # State is locked so don't do anything
+            return
+        if not self.env.user.get_user_warehouse().u_auto_assign_batch:
+            # Only do this if configured to
+            return
+
+        for batch in self:
+            if batch.state in ["waiting", "in_progress"]:
+                picks_to_assign = batch.picking_ids.filtered(
+                    lambda p: p.state == "confirmed"
+                    and p.mapped("move_lines").filtered(
+                        lambda move: move.state not in ("draft", "cancel", "done")
+                    )
+                )
+                if picks_to_assign:
+                    picks_to_assign.with_context(lock_batch_state=True).action_assign()
```

Comment:
If all the pickings are successfully assigned, the state of the batch should change to ready if it was waiting?